test/pobox_give_away_SUITE.erl

```diff
@@ -0,0 +1,88 @@
+-module(pobox_give_away_SUITE).
+-include_lib("common_test/include/ct.hrl").
+-compile(export_all).
+
+all() -> [
+  owner_gives_away_ownership,
+  owner_gives_away_ownership_with_data,
+  owner_gives_away_ownership_to_dead_process,
+  foreign_process_tries_to_gives_away_ownership,
+  give_away_does_not_change_heir
+].
+
+owner_gives_away_ownership(_Config) ->
+    {ok, Box} = pobox:start_link([]),
+    Self = self(),
+    Func = fun() ->
+      Fn = fun(F) -> receive X -> Self ! X end, F(F) end,
+      Fn(Fn)
+    end,
```

Comment:
You can instead use a named fun:

```erlang
Func = fun F() -> receive X -> Self ! X end, F() end
```
`F` will only be in scope within the fun.

Reply:
That was my first implementations but I changed it trying to support older Erlang versions. I guess since we are using gen_statem I might as well switch it back.

Reply:
Done.